tests/codegen/scratch_buffer.cpp

```diff
@@ -115,6 +115,20 @@ TEST(codegen, variable_stack)
       LARGE_ON_STACK_LIMIT);
 }
 
+TEST(codegen, map_key_scratch_buf)
+{
+  test_stack_or_scratch_buffer("kprobe:f { @x[1] = 1; @y[\"yyyy\"] = @x[1]; }",
+                               NAME,
+                               SMALL_ON_STACK_LIMIT);
+}
+
+TEST(codegen, map_key_stack)
```

Comment:
Is it worth adding codegen tests for some of the function calls e.g. `hist` or `has_key` ?